third_party/maya/lib/usdMaya/xformStack.h

```diff
@@ -0,0 +1,385 @@
+    ///
+    /// Consists of these xform operators:
+    ///    translate
+    ///    rotatePivotTranslate
+    ///    rotatePivot
+    ///    rotate
+    ///    rotateAxis
+    ///    rotatePivot^-1 (inverted twin)
+    ///    scalePivotTranslate
+    ///    scalePivot
+    ///    shear
+    ///    scale
+    ///    scalePivot^-1 (inverted twin)
+    PXRUSDMAYA_API
+    static const PxrUsdMayaXformStack& MayaStack();
+
+
+    /// \brief The Common API xform stack
+    ///
+    /// Consists of these xform operators:
+    ///    translate
+    ///    pivot
+    ///    rotate
+    ///    scale
+    ///    pivot^-1 (inverted twin)
+    PXRUSDMAYA_API
+    static const PxrUsdMayaXformStack& CommonStack();
+
+    /// \brief xform "stack" consisting of only a single matrix xform
+    ///
+    /// This stack will match any list of xform ops that consist of a single
+    /// matrix "transform" op, regardless of name.
+    /// Consists of these xform operators:
+    ///    transform
+    PXRUSDMAYA_API
+    static const PxrUsdMayaXformStack& MatrixStack();
+
+    /// \brief dummy recursion endpoint for FirstMatchingSubstack variadic template
+    PXRUSDMAYA_API
+    inline static std::vector<OpClassPtr>
+    FirstMatchingSubstack(
+            const std::vector<UsdGeomXformOp>& xformops,
+            MTransformationMatrix::RotationOrder* MrotOrder=nullptr)
+    {
+        return std::vector<OpClassPtr>();
+    }
+
+   /// \brief Runs MatchingSubstack against the given list of stacks
+    ///
+    /// Returns the first non-empty result it finds; if all stacks
+    /// return an empty vector, an empty vector is returned.
+    template<typename... RemainingTypes>
+    static std::vector<OpClassPtr>
+    FirstMatchingSubstack(
+            const std::vector<UsdGeomXformOp>& xformops,
+            MTransformationMatrix::RotationOrder* MrotOrder,
+            const PxrUsdMayaXformStack& firstStack,
+            RemainingTypes&... otherStacks)
+    {
+        // WARNING: this logic is currently duplicated in wrapXformStack.cpp,
+        // Usd_PyXformStack::FirstMatchingSubstack, because it was much
+        // simpler / readable than using variadic templates to convert a
+        // python tuple into variadic arguments. Should be ok because the
+        // logic is currently so simple. If this logic changes, either change it
+        // in both places, or modify the python wrapper to call the variadic
+        // template function directly.
+        if (xformops.empty()) return std::vector<OpClassPtr>();
+
+        std::vector<PxrUsdMayaXformStack::OpClassPtr> stackOps = \
+                firstStack.MatchingSubstack(xformops, MrotOrder);
+        if (!stackOps.empty())
+        {
+            return stackOps;
+        }
+        return FirstMatchingSubstack(xformops, MrotOrder, otherStacks...);
+    }
+
+private:
+    PxrUsdMayaXformStack(
+            const OpClassList ops,
+            const std::vector<std::pair<size_t, size_t> > inversionTwins,
```

Comment:
Did you mean to pass a const-ref in here instead?

Reply:
Oops, yes, that should be a const-ref. Good catch.